lib/Dialect/SV/Transforms/PrettifyVerilog.cpp

```diff
@@ -52,6 +55,46 @@ static bool isVerilogUnaryOperator(Operation *op) {
   return false;
 }
 
+/// Sink an operation into the same block where it is used.  This will clone the
+/// operation so it can be sunk into multiple blocks. If there are no more uses
+/// in the current block, the op will be removed.
+void PrettifyVerilogPass::sinkOpToUses(Operation *op) {
```

Comment:
Drive-by comment: some operations should not (logically) be cloned and pushed around. Is there some op trait you can assert on? (For future idiot proofing.)

Reply:
added an `assert(hasNoEffect(op))`.